<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Creating Promises!</title>
</head>
<body>
<script>
  const p = new Promise((resolve, reject) => { //either the promise resolves or it can reject itself due to an error, data issues, etc. 
    setTimeout(() => {
      resolve('Data feedback returned');
      //reject(Error('Err Data did not return')); //to have the error return on the line where the issue was, not where the catch was. 
    }, 1000);
  });

  p
    .then(data => {
      console.log(data);
    })
    .catch(err => {
      console.error(err);
    });


</script>
</body>
</html>
